src/subl.ts

```
@@ -1,10 +1,83 @@
+const generateProjects: Fig.Generator = {
+  template: "filepaths",
+  filterTemplateSuggestions: (suggestions) => {
+    const out: Fig.Suggestion[] = [];
+    for (const suggestion of suggestions) {
+      if (suggestion.type === "folder") {
+        out.push(suggestion);
+        continue;
+      }
+      if (/^\.sublime-(project|workspace)$/.test(suggestion.name)) {
+        suggestion.priority = 75;
+        out.push(suggestion);
+      }
+    }
+    return out;
+  },
+};
+
 const completionSpec: Fig.Spec = {
   name: "subl",
   description: "Sublime Text",
   args: {
+    name: "files|directories",
     isVariadic: true,
+    isOptional: true,
     template: ["folders", "filepaths"],
   },
+  options: [
+    {
+      name: "--project",
+      description: "Load the given project",
+      args: {
+        name: "project",
+        generators: generateProjects,
+      },
+    },
+    {
+      name: "--command",
+      description: "Run the given command",
+      args: {
+        name: "command",
+      },
+    },
+    {
+      name: ["-n", "--new-window"],
+      description: "Open a new window",
+    },
+    {
+      name: ["-a", "--add"],
+      description: "Add folders to the current window",
+    },
```

@QuiiBz -n and -a aren't mutually exclusive, but when -n is present -a is silently ignored. Should these be marked as `exclusiveOn`?

---

Totally! You can add `-a` to `exclusiveOn` for the `"-n", "--new-window"` option